dialog/src/main/java/roboy/logic/InferenceEngine.java

```diff
@@ -84,4 +88,25 @@
      */
     UtteranceSentiment inferSentiment(Interpretation input);
 
+    /**
+     * Basic inference method
+     * Infers the snapchat filter of the user's input
+     * tries to extract and ground the information from the available Interpretation
+     *
+     * @param input,existingFilterMap
+     * @return List containing inferred result/null if NA
+     */
+    List<String> inferSnapchatFilter(Interpretation input, Map<String,List<String>> existingFilterMap);
```

Comment:
InferenceEngine is a very high level interface and shall not contain inferSnapchat or game methods. they should be local for Inference you are using or even located in corresponding states.

Reply:
regarding the snapchat names: I meant to store exact filter names how the python side is expecting it, not all possible synonyms, this is handle inside dialog.

Reply:
regarding the inference, I meant meant having them inside Inference we are using, not the top level interface class InferenceEngine. now every time someone wants to implement their own Inference login they have to have your methods implemented which doesnt make sense